### DISCUSSION TITLE
`sbtPluginPublishLegacyMavenStyle := false` and scripted test

### DISCUSSION THREAD

**Discussion by xuwei-k** [sbt/sbt#7277] [Q&A]
Could not resolve local sbt plugin if `sbtPluginPublishLegacyMavenStyle := false`.
Is there any solution with keep `sbtPluginPublishLegacyMavenStyle := false` ? Should we use local maven repo? 🤔 

- https://github.com/xuwei-k/sbtPluginPublishLegacyMavenStyle-false-scripted-test/actions/runs/5162314405/jobs/9299928807
- https://github.com/xuwei-k/sbtPluginPublishLegacyMavenStyle-false-scripted-test/commit/518f6e474b55980de488e16342e9b2690d9a9675

```
[info] 	delivering ivy file to /home/runner/work/sbtPluginPublishLegacyMavenStyle-false-scripted-test/sbtPluginPublishLegacyMavenStyle-false-scripted-test/target/scala-2.12/sbt-1.0/ivy-0.1.0-SNAPSHOT.xml
[info] 	published foo_2.12_1.0 to /home/runner/.ivy2/local/com.example/foo/scala_2.12/sbt_1.0/0.1.0-SNAPSHOT/jars/foo_2.12_1.0.jar
[info] 	published foo_2.12_1.0 to /home/runner/.ivy2/local/com.example/foo/scala_2.12/sbt_1.0/0.1.0-SNAPSHOT/srcs/foo_2.12_1.0-sources.jar
[info] 	published foo_2.12_1.0 to /home/runner/.ivy2/local/com.example/foo/scala_2.12/sbt_1.0/0.1.0-SNAPSHOT/docs/foo_2.12_1.0-javadoc.jar
[info] 	published foo_2.12_1.0 to /home/runner/.ivy2/local/com.example/foo/scala_2.12/sbt_1.0/0.1.0-SNAPSHOT/poms/foo_2.12_1.0.pom
[info] 	published ivy to /home/runner/.ivy2/local/com.example/foo/scala_2.12/sbt_1.0/0.1.0-SNAPSHOT/ivys/ivy.xml
```

```
[info] [error] lmcoursier.internal.shaded.coursier.error.FetchError$DownloadingArtifacts: Error fetching artifacts:
[info] [error] file:///home/runner/.ivy2/local/com.example/foo/scala_2.12/sbt_1.0/0.1.0-SNAPSHOT/jars/foo.jar: not found: /home/runner/.ivy2/local/com.example/foo/scala_2.12/sbt_1.0/0.1.0-SNAPSHOT/jars/foo.jar
[info] [error] 
[info] [error] 	at lmcoursier.internal.shaded.coursier.Artifacts$.$anonfun$fetchArtifacts$9(Artifacts.scala:365)
[info] [error] 	at lmcoursier.internal.shaded.coursier.util.Task$.$anonfun$flatMap$extension$1(Task.scala:14)
```

**Comment by eed3si9n** [comment]
sbt 1.x's scripted test relies only on `publishLocal`, which is Ivy repo publishing, so I think we should keep using the Ivy repo.
I am guessing that `sbtPluginPublishLegacyMavenStyle := false` is inadvertently affecting the Ivy repo publishing, likely by listing `foo.jar` in `ivy.xml` but publishing `foo_2.12_1.0.jar`. Since Ivy is capable of representing extra attribute fine, I don't think `sbtPluginPublishLegacyMavenStyle` should have affect on `publishLocal`.

@adpi2 I think we should convert this discussion into an issue.

**Reply by adpi2** [reply]
Moved to https://github.com/sbt/sbt/issues/7285
